src/error.rs

```diff
@@ -32,14 +32,16 @@ pub enum Error {
     MissingCodec,
     #[error("Validation error {0}")]
     Validation(String),
+    #[error("Acquire lock error {0}")]
+    LockError(#[from] local_sync::semaphore::AcquireError),
```

Comment:
What's this?

Reply:
For the native HTTP/2 connector I didn't want to use a Tokio::Mutex. Instead I used local_sync's semaphore with a single permit. Will change it to local_sync's mutex eventually

Reply:
I didn't found where create this error?

Reply:
``` Rust
     
          if connect_to_h2 {
            let lock = {
                let connecting = unsafe { &mut *self.connecting.get() };
                let lock = connecting
                    .entry(key.clone())
                    .or_insert_with(|| Rc::new(local_sync::semaphore::Semaphore::new(1)));
                lock.clone()
            };

            // get lock and try again
            let _guard = lock.acquire().await?;   <<<< **Returns AcquireError here**
            if let Some(conn) = try_get!(self, h2_pool, key) {
                return Ok(conn.into());
            }

```